geomesa-accumulo/geomesa-accumulo-jobs/src/main/scala/org/locationtech/geomesa/jobs/accumulo/mapreduce/package.scala

```diff
@@ -0,0 +1,136 @@
+/***********************************************************************
+ * Copyright (c) 2013-2021 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.jobs.accumulo
+
+import java.io.{BufferedOutputStream, DataInput, DataOutput, PrintStream}
+import java.nio.charset.StandardCharsets
+import java.util.{Base64, Scanner}
+
+import com.github.benmanes.caffeine.cache.{CacheLoader, Caffeine}
+import org.apache.accumulo.core.data.Key
+import org.apache.hadoop.conf.{Configurable, Configuration}
+import org.apache.hadoop.fs.Options.CreateOpts
+import org.apache.hadoop.fs.{CreateFlag, FileContext, FileSystem, Path}
+import org.apache.hadoop.io.{BinaryComparable, Text, Writable, WritableComparable}
+import org.apache.hadoop.mapreduce.{Job, Partitioner}
+import org.locationtech.geomesa.utils.io.WithClose
+
+import scala.collection.mutable.ArrayBuffer
+
+package object mapreduce {
+
+  object Configurator {
+
+    private val TypeNameKey   = "org.locationtech.geomesa.accumulo.typename"
+    private val PartitionsKey = "org.locationtech.geomesa.accumulo.partitions"
+
+    def setTypeName(conf: Configuration, typeName: String): Unit = conf.set(TypeNameKey, typeName)
+    def getTypeName(conf: Configuration): String = conf.get(TypeNameKey)
+    def setPartitions(conf: Configuration, partitions: Seq[String]): Unit =
+      conf.set(PartitionsKey, partitions.mkString(","))
+    def getPartitions(conf: Configuration): Option[Seq[String]] = Option(conf.get(PartitionsKey)).map(_.split(","))
+  }
+
+  class TableAndKey extends WritableComparable[TableAndKey] {
+
+    private var table: Text = _
+    private var key: Key = _
+
+    def this(table: Text, key: Key) = {
+      this()
+      this.table = table
+      this.key = key
+    }
+
+    def getTable: Text = table
+    def setTable(table: Text): Unit = this.table = table
+    def getKey: Key = key
+    def setKey(key: Key): Unit = this.key = key
+
+    override def write(out: DataOutput): Unit = {
+      table.write(out)
+      key.write(out)
+    }
+
+    override def readFields(in: DataInput): Unit = {
+      table = new Text()
+      table.readFields(in)
+      key = new Key()
+      key.readFields(in)
+    }
+
+    override def compareTo(o: TableAndKey): Int = {
```

Comment:
Do we need to do anything with equals?  Or is that just a nice-to-have?

Reply:
ah, i think it doesn't matter. it needs to be comparable so that the reducer can sort the output. but maybe it would be good to also implement equals and hash code...

Reply:
Your call.

Reply:
i probably should have gone back and done this but didn't get around to it... hopefully it won't bite us in some subtle way :P

Reply:
Yeah, I don't think this'll be our shared mistake this year.....  If I did, I wouldn't have hit the merge button.  The docs on compareTo make it sound optional to think about equals/hashCode/whatever.